xlators/protocol/client/src/client-handshake.c

```diff
@@ -644,7 +644,7 @@ client_post_handshake(call_frame_t *frame, xlator_t *this)
         list_for_each_entry_safe(fdctx, tmp, &conf->saved_fds, sfd_pos)
         {
             if (fdctx->remote_fd != -1 ||
-                (!list_empty(&fdctx->lock_list) && conf->strict_locks))
+                (!fdctx_lock_lists_empty(fdctx) && conf->strict_locks))
```

Comment:
I went through the code again. I didn't realize lk_ctx is a global list. If there is already a global list, do we ever need a local list per client? It looks like this code doesn't have the bugs we fixed in client too. Maybe we can delete local lock list completely? If yes https://github.com/gluster/glusterfs/issues/2483 will be completely fixed. Right? cc @itisravi

Reply:
@pranithk  Both client-lk.c  and fd-lk.c seem to have lot of code originally introduced by commit f764516c2e526624ce0088963924ff2d88304553 which is not needed because we don't do lock healing in client xlator. But I think it is better to have this fix in (with other review comments addressed) and do the clean up as a part of 2483 (I will be picking it up soon).

Reply:
Does that seem reasonable?

Reply:
I am guessing you guys are under some deadline. Sure. Works.